pallets/liquidity-provider/src/lib.rs

```diff
@@ -3,18 +3,29 @@
 use frame_support::{decl_error, decl_event, decl_module, dispatch};
 use frame_system::ensure_signed;
 use orml_traits::{MultiCurrency, MultiReservableCurrency};
-use valiu_node_commons::Asset;
+use valiu_node_commons::{Asset, Collateral};
 
 #[cfg(test)]
 mod mock;
 #[cfg(test)]
 mod tests;
 
-type BalanceOf<T> =
-    <<T as Trait>::Currency as MultiCurrency<<T as frame_system::Trait>::AccountId>>::Balance;
+type CollateralBalance<T> =
+    <<T as Trait>::Collateral as MultiCurrency<<T as frame_system::Trait>::AccountId>>::Balance;
+type MintMembers = pallet_membership::Instance0;
+type ProviderMembers = pallet_membership::DefaultInstance;
 
-pub trait Trait: pallet_membership::Trait {
-    type Currency: MultiReservableCurrency<Self::AccountId, CurrencyId = Asset>;
+pub trait Trait:
+    pallet_membership::Trait<MintMembers> + pallet_membership::Trait<ProviderMembers>
+{
+    type Asset: MultiCurrency<
+        Self::AccountId,
+        Balance = <<Self as Trait>::Collateral as MultiCurrency<
+            <Self as frame_system::Trait>::AccountId,
+        >>::Balance,
+        CurrencyId = Asset,
+    >;
+    type Collateral: MultiReservableCurrency<Self::AccountId, CurrencyId = Asset>;
```

Comment:
The `token` module expects a single `CurrencyId` but in the near future, `Collateral` should be `MultiReservableCurrency<Self::AccountId, CurrencyId = Collateral>` somehow.